src/seedsigner/views/screensaver.py

```diff
@@ -183,50 +187,58 @@ def start(self):
         # Store the current screen in order to restore it later
         self.last_screen = self.renderer.canvas.copy()
 
-        screensaver_start = int(time.time() * 1000)
-
         # Screensaver must block any attempts to use the Renderer in another thread so it
         # never gives up the lock until it returns.
         with self.renderer.lock:
             try:
+                last_update_time = time.time()
+                current_speed = self.rand_speed() * self.pixels_per_second
+                
                 while self._is_running:
                     if self.buttons.has_any_input() or self.buttons.override_ind:
                         break
 
+                    current_time = time.time()
+                    elapsed_time = current_time - last_update_time
+                    
+                    # Calculate pixel movement based on elapsed time and target speed
+                    pixels_to_move = elapsed_time * current_speed
```

Comment:
I would cast to `int` here. The `self.cur_x`/`cur_y` internal vars should never get float values, even if only temporarily.

Reply:
For very fast hardware and a random very slow movement speed, it's possible that this value will round to zero.

I would check for that and `continue` and let `last_update_time` accumulate more time.